File: isma-intg-lib/isma-intg-lib-rk31/src/main/java/ru/nstu/isma/intg/lib/rungeKutta/rk31/internal/Rk31Stage3Calculator.java
package ru.nstu.isma.intg.lib.rungeKutta.rk31.internal;

import ru.nstu.isma.intg.api.methods.StageCalculator;

/**
 * @author Dmitry Dostovalov
 * @since 16.10.15
 */

public class Rk31Stage3Calculator extends StageCalculator {
    @Override
    public double yk(double step, double y, double f, double[] stages) {
        return y + 11.0/8.0 * stages[0] - 3.0/8.0 * stages[1];
    }

    @Override
    public double k(double step, double y, double f, double[] stages, double stagesY, double stagesF) {
        return step * stagesF;
    }
}

File: isma-intg-lib/isma-intg-lib-rk31/src/main/kotlin/ru/nstu/isma/intg/lib/rungeKutta/rk31/internal/Rk31Stage3Calculator.kt
package ru.nstu.isma.intg.lib.rungeKutta.rk31.internal

import ru.nstu.isma.intg.api.methods.StageCalculator

/**
 * @author Dmitry Dostovalov
 * @since 16.10.15
 */
class Rk31Stage3Calculator : StageCalculator() {
    override fun yk(step: Double, y: Double, f: Double, stages: DoubleArray): Double {
        return y + 11.0 / 8.0 * stages[0] - 3.0 / 8.0 * stages[1]
    }

    override fun k(step: Double, y: Double, f: Double, stages: DoubleArray, stagesY: Double, stagesF: Double): Double {
        return step * stagesF
    }
}
